contracts/Staking.sol

```diff
@@ -147,10 +151,11 @@ contract Staking is IStaking, Ownable {
                 (amountToTransfer * (100 - stakingInfo.feePercentage)) /
                 100;
         }
-        rewards[msg.sender] = 0;
         uint256 balanceRewardToken = rewardToken.balanceOf(address(this));
-        if (balanceRewardToken < amountToTransfer)
+        if (balanceRewardToken < amountToTransfer) {
             amountToTransfer = balanceRewardToken;
+            rewards[msg.sender] -= amountToTransfer;
+        } else rewards[msg.sender] = 0;
```

Comment:
you can just after if {}  use next rewards[msg.sender] -= amountToTransfer; and it's will corecty work. or i mistake?